### ISSUE TITLE
load custom classifier with batch_run

### ISSUE THREAD

**Issue by sonjablumenstock** [MouseLand/suite2p#513]
Hi,

I'm running suite2p in batch using a script that loops through a file list. I'm wondering how I can supply a path to load my own classifier (my_classifier.npy)?
I tried giving a 'classfile' key with a path value to the ops and db dictionaries in run_s2p, but that was not the right solution.

Thanks so much in advance,
Sonja

**Comment by chriski777** [comment]
Hey @sonjablumenstock, unfortunately as of now, there is no ops param you can pass into use your own classifier. This feature though is something we've implemented and it should be available with our new release(should happen soon). You'd be able to set your classifier path with `ops['classifier_path'] = 'path_to_blahblah'`.


If it is urgent, what you can do is use the GUI and set your default classifier to your own classifier. This will change the default $HOME/.suite2p/classifiers/classifier_user.npy' you have which in turn would cause your own classifier to be used during your script. Then, after your analysis is complete, you can reset the classifier to the built-in one with the GUI.

**Comment by sonjablumenstock** [comment]
Thanks for the info!
In my case it makes sense to wait for the new release, as I have too recordings to process 'manually' (and need different classifiers). When will the new version be released?

**Comment by nickdelgrosso** [comment]
Should be this week.  But you can also have it now--it's a little more work, but you can clone from our "dev" branch and install to get the current version:

```
git clone https://github.com/MouseLand/suite2p
cd suite2p
git checkout -b dev
git pull origin dev
conda activate suite2p  # your existing suite2p conda environment
python setup.py install
```

**Comment by sonjablumenstock** [comment]
Thanks for your answers and your support.

Using run_s2p, still the own classifier_path assigned to ops_classifile is not used. 
Instead, suite2p always uses the user_classfile (from classifier_user.npy, which is saved in the home directory). I think the reason is  in the following statement, that gives user_classfile a higher priority (if present, which it always is). 

https://github.com/MouseLand/suite2p/blob/c7267e19990666272de25bd313979fcee35188a1/suite2p/run_s2p.py#L197-L212

My workaround would be to overwrite user_classifier, but as I'll need multiple different classifiers for my analysis, it would be great to have an option to safely assign a custom path in the future.

**Comment by nickdelgrosso** [comment]
Great catch, @sonjablumenstock .  Agreed, the "classifier_path" argument should override the other classfile-related settings.

**Comment by nickdelgrosso** [comment]
@sonjablumenstock , it looks like this is merged.  If you update your suite2p (pull from dev), the "classifier_path" argument should work.

**Comment by nickdelgrosso** [comment]
Okay, this is included in the suite2p 0.19 release.  Closing the issue.
